### ISSUE TITLE
Bitwise operation weirdness

### ISSUE THREAD

**Issue by benkuhn** [pydata/pandas#9016]
It seems like `pd.Series([x]) | pd.Series([y])` with `x, y` integers returns `pd.Series([x | y]).astype(bool)`. This is a reasonable semantic, but `pd.Series([x]) & pd.Series([y])` seems to return `pd.Series([x & y % 2]) == 1`, which is a lot weirder. Is there a justification for this? I couldn't find one in the documentation (`&` is hard to search!), so it may be a bug.

**Comment by tvyomkesh** [comment]
Tried a quick & dirty hack and getting back desired behavior. Tests passed too. Sharing snippet so I can gather some expert comments and advise. Would be happy to work on a pull request if this looks in the correct direction. Thanks.
```
diff --git a/pandas/core/ops.py b/pandas/core/ops.py
index 46429cc..59be639 100644
--- a/pandas/core/ops.py
+++ b/pandas/core/ops.py
@@ -657,10 +657,16 @@ def _bool_method_SERIES(op, name, str_rep):
         if isinstance(other, pd.Series):
             name = _maybe_match_name(self, other)

-            other = other.reindex_like(self).fillna(False)
-            return self._constructor(na_op(self.values, other.values),
+            if other.dtype == self.dtype and self.dtype == 'int64':
+                other = other.reindex_like(self).fillna(False)
+                return self._constructor(na_op(self.values, other.values),
                                      index=self.index,
                                      name=name).fillna(False)
+            else:
+                other = other.reindex_like(self).fillna(False).astype(bool)
+                return self._constructor(na_op(self.values, other.values),
+                                     index=self.index,
+                                     name=name).fillna(False).astype(bool)
         elif isinstance(other, pd.DataFrame):
             return NotImplemented
         else:
```

Some outputs showing results after the above fix

```
In [2]: a = Series([True, False, True], list('bca'))
In [3]: b = Series([])
In [6]: a & b
Out[6]:
b    False
c    False
a    False
dtype: bool

In [7]: c = Series(range(10))
In [8]: d = Series([15] * 10)
In [11]: c & d
Out[11]:
0    0
1    1
2    2
3    3
4    4
5    5
6    6
7    7
8    8
9    9
dtype: int64
```
Thanks,

**Comment by jreback** [comment]
this should be handled in ``na_op``, with the fill value determined by ``dtype``

**Comment by tvyomkesh** [comment]
@jreback thanks. I will look into this and try and work on a PR.

**Comment by tvyomkesh** [comment]
@jreback for now, I have made the changes in wrapper() itself instead of na_op() because it looked to me like wrapper is controlling the input and output fill and dtype. Once wrapper() takes care of this, na_op() seems to be doing the right thing by itself and so I did not have to change anything to get back the expected behavior. Happy to change if things need to be altered toward better design etc. Thanks.
